### PR TITLE
Add servlets that query and insert data in verifications table 

### PR DESCRIPTION
**Pull request by vahinim**
Add SpannerClient.java
Add VerificationDatabaseHelper.java
Add GetStoresVerificationServlet.java, that fetches data from verifications table
Add AddStoreVerificationStatusServlet.java, that inserts verification data into verifications table

### CODE REVIEW

File: FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/AddStoreVerificationStatusServlet.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+@WebServlet(value = "/verifications/new")
+public class AddStoreVerificationStatusServlet extends HttpServlet {
+
+    private Gson gson = new Gson();
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        String json = request.getReader().readLine();
```
Comment (krati-tiwari) [inline FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/AddStoreVerificationStatusServlet.java]: rename to verificationRequest?

File: FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/GetStoreVerificationsServlet.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+        ResultSet verificationData = VerificationDatabaseHelper.queryData(output);
+
+        if (verificationData.next()) {
+            int columnAgentEmailIndex = verificationData.getColumnIndex(VerificationDatabaseHelper.COLUMN_AGENT_EMAIL);
+            int columnStorePhoneIndex = verificationData.getColumnIndex(VerificationDatabaseHelper.COLUMN_STORE_PHONE);
+            int columnVerificationLatitudeIndex = verificationData.getColumnIndex(VerificationDatabaseHelper.COLUMN_VERIFICATION_LATITUDE);
+            int columnVerificationLongitudeIndex = verificationData.getColumnIndex(VerificationDatabaseHelper.COLUMN_VERIFICATION_LONGITUDE);
+            int columnVerificationStatusIndex = verificationData.getColumnIndex(VerificationDatabaseHelper.COLUMN_VERIFICATION_STATUS);
+            int columnVerificationTimeIndex = verificationData.getColumnIndex(VerificationDatabaseHelper.COLUMN_VERIFICATION_TIME);
```
Comment (krati-tiwari) [inline FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/GetStoreVerificationsServlet.java]: Make each statement in 2 lines for better readability.

File: FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/VerificationDatabaseHelper.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    public static boolean writeData(Verification newVerification, PrintWriter pw) {
+        Mutation verificationMutation;
+        verificationMutation = Mutation.newInsertBuilder(TABLE_NAME)
+                .set(COLUMN_AGENT_EMAIL).to(newVerification.getAgentEmail())
+                .set(COLUMN_STORE_PHONE).to(newVerification.getStorePhone())
+                .set(COLUMN_VERIFICATION_LATITUDE).to(newVerification.getVerificationCoordinates().getLatitude())
+                .set(COLUMN_VERIFICATION_LONGITUDE).to(newVerification.getVerificationCoordinates().getLongitude())
+                .set(COLUMN_VERIFICATION_STATUS).to(newVerification.getStoreVerificationStatus())
+                .set(COLUMN_VERIFICATION_TIME).to(Timestamp.of(java.sql.Timestamp.valueOf(ZonedDateTime.now(ZoneId.of("Asia/Kolkata")).toLocalDateTime())))
```
Comment (krati-tiwari) [inline FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/VerificationDatabaseHelper.java]: Split in 2 line

File: FosServer/src/main/java/com/example/fostracker/models/Agent.java [hunk 4]
```diff
@@ -27,6 +29,21 @@ public Agent(Name name, String email, String phone, Coordinates coordinates) {
         this.agentCreationDateTime = Timestamp.valueOf(ZonedDateTime.now(ZoneId.of("Asia/Kolkata")).toLocalDateTime());
     }
 
+    /**
+     * @param name                  - The agent's name
+     * @param email                 - the gmail address associated with the agent's account
+     * @param phone                 - The phone number associated with the agent's account
+     * @param coordinates           - The live coordinates of the agent at the time of creation. They can be updated later
```
Comment (krati-tiwari) [inline FosServer/src/main/java/com/example/fostracker/models/Agent.java:36]: keep same as above params

**Review by krati-tiwari — CHANGES_REQUESTED** [review]

File: FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/GetStoreVerificationsServlet.java (hunk 2 above)
Comment (vahinim) [reply FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/GetStoreVerificationsServlet.java]: done

File: FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/VerificationDatabaseHelper.java (hunk 3 above)
Comment (vahinim) [reply FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/VerificationDatabaseHelper.java]: done

File: FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/AddStoreVerificationStatusServlet.java (hunk 1 above)
Comment (vahinim) [reply FosServer/src/main/java/com/example/fostracker/servlets/VerificationServlet/AddStoreVerificationStatusServlet.java]: done

File: FosServer/src/main/java/com/example/fostracker/models/Agent.java (hunk 4 above)
Comment (vahinim) [reply FosServer/src/main/java/com/example/fostracker/models/Agent.java:36]: Can you elaborate it a little more ? The second constructor comments are same as first one with last line changed to reflect the agentCreationDateTime argument. And this constructor is used while querying, so we will not set the present time to Agent creation time while querying.

File: FosServer/src/main/java/com/example/fostracker/servlets/HelloAppEngine.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+package com.example.fostracker.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(name = "HelloAppEngine", value = "/hello")
```
Comment (trblessy) [inline FosServer/src/main/java/com/example/fostracker/servlets/HelloAppEngine.java:25]: Do we need this servlet now?

**Review by trblessy — CHANGES_REQUESTED** [review]

**Review by trblessy — APPROVED** [review]

**Review by krati-tiwari — APPROVED** [review]

File: FosServer/src/main/java/com/example/fostracker/servlets/HelloAppEngine.java (hunk 5 above)
Comment (vahinim) [reply FosServer/src/main/java/com/example/fostracker/servlets/HelloAppEngine.java:25]: removed

**Pull request merged**